### PR TITLE
[DX] Add ->withPhpXSets() for PHP 7.4 and lower, improve ->withPreparedSets()

### PR DESCRIPTION
**Pull request by TomasVotruba**
The `withPhpSets()` method uses named arguments, available since PHP 8.0.
If you use Rector on PHP 7.4 and lower, it can be confusing to use:

```php
return RectorConfig::configure()
     ->withPhpSets(false, false, false, true);
```

:x: 

That's unreadable and error prone. Once new version or order is changed, it behaves differently. 

<br>

Instead, this PR introduces clear method API:

```php
return RectorConfig::configure()
     ->withPhp73Sets();
```

:heavy_check_mark:  

Nice and clear :)

<br>

Ideally, the `withPhpSets()` should be used without any argument, so it always follows your `composer.json` version :+1:

### CODE REVIEW

File: src/functions/notifications.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+// @use in config builder
+function reportMethodUsedAbovePHP80(string $calledMethod, string $recommendedMethod): void
```
Comment (samsonasik) [inline src/functions/notifications.php]: this needs to be registered in scoper.php

https://github.com/rectorphp/rector-src/blob/61a419fe248538b2559ef1df804702e1d4392dee/scoper.php#L50

File: src/functions/notifications.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+{
+    // current project version check
+    if (PHP_VERSION_ID < 80000) {
+        return;
+    }
+
+    echo sprintf(
```
Comment (samsonasik) [inline src/functions/notifications.php]: Can symfonySyle->warning with yellow color used here?

File: src/functions/notifications.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
```
Comment (samsonasik) [inline src/functions/notifications.php]: here should be `function_exists()` like other function definition:

```suggestion
if (! function_exists('reportMethodUsedAbovePHP80')) {
```

**Review by samsonasik — CHANGES_REQUESTED** [review]
`RemovePhpVersionIdCheckRector` should be skipped for Notifier file

**Comment by TomasVotruba** [comment]
Ready for next review :+1:  :)

**Review by samsonasik — APPROVED** [review]
LGTM 👍

**Comment by TomasVotruba** [comment]
Thank you :clap:

File: src/Configuration/RectorConfigBuilder.php [hunk 4]
```diff
@@ -469,6 +478,82 @@ public function withPhpSets(
         return $this;
     }
 
+    // suitable for PHP 7.4 and lower, before named args
+    public function withPhp53Sets(): self
+    {
+        Notifier::notifyNotSuitableMethodForPHP80(__METHOD__, 'withPhpSets');
```
Comment (samsonasik) [inline src/Configuration/RectorConfigBuilder.php:484]: The warning seems shown multiple times due to **parallel**:

```                                                                           
 [WARNING] The "Rector\Configuration\RectorConfigBuilder::withPhp53Sets()"      
           method is suitable for PHP 7.4 and lower. Use "withPhpSets()" method 
           instead.                                                             
                                                                                

  0/99 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   0%
                                                                                
 [WARNING] The "Rector\Configuration\RectorConfigBuilder::withPhp53Sets()"      

           method is suitable for PHP 7.4 and lower. Use "withPhpSets()" method 
           instead.                                                      
                                                                                
 [WARNING] The "Rector\Configuration\RectorConfigBuilder::withPhp53Sets()"      
           method is suitable for PHP 7.4 and lower. Use "withPhpSets()" method 
           instead.                                                             
                                                                              
                                                                                
 [WARNING] The "Rector\Configuration\RectorConfigBuilder::withPhp53Sets()"      
           method is suitable for PHP 7.4 and lower. Use "withPhpSets()" method 
           instead.                                                        
```

that can't be tweak here, only on `ApplicationFileProcessor` level, the single warning can be shown...

Also, I am thinking if github workflow uses:

```
      matrix:
        php-versions: ['7.4, '8.0']
```

that may not suit, so probably suggest to use `withSets()`...

**Pull request merged**